docs/chain-details/module_slashing.md

```diff
@@ -0,0 +1,137 @@
+### `slashing` module
+
+### Introduction
+
+Validators are responsible for signing or proposing block at each consensus round. A penalty should be imposed on validators' misbehavior to reinforce this.
+
+Specifically, `slashing` functionality that aims to dis-incentivize network-observable actions, such as faulty validations. The penalties may include losing some amount of their stake, losing their ability to perform the network functionality for a period of time, collect rewards etc.
+
+### Overview
+
+#### Network parameters
+
+Below are all the network parameters used to configure the behavior of validator punishments. Details of all these parameters and their effect on behavior of validator punishments is discussed later in this document.
+
+- `signed_blocks_window`: Number of blocks for which the liveness is calculated for uptime tracking;
+- `min_signed_per_window`: Maximum percentage of blocks with faulty/missed validations allowed for an account in last; `signed_blocks_window` blocks before it gets deactivated;
+- `downtime_jail_duration`: Duration for [jailing](#jailing);
+-  `slash_fraction_double_sign`: Percentage of funds being slashed when validator makes a byzantine fault; and
+-  `slash_fraction_downtime`: Percentage of funds being slashed when a validator is non-live.
+
+#### Slashing mechanism
+
+Punishments for a validator are triggered when they either make a _byzantine fault_ or become _non-live_:
+
+- Liveness Faults (Low availability)
+
+  A validator is said to be **non-live** when they fail to sign at least `min_signed_per_window` blocks (in percentage) in the
+  last `signed_blocks_window` blocks successfully. `signed_blocks_window` and `min_signed_per_window` are network
+  parameters and can be configured during genesis and can be updated during runtime by the governance module.
+
+:::tip Example:
+For example, if `block_signing_window` is `2000` blocks and `min_signed_per_window` is `0.5`, a validator will
+be marked as **non-live** and jailed if they fail to successfully sign at least `2000*0.5=1000` blocks in last `2000` blocks.
+:::
+
+- Byzantine Faults (Double signing)
```

Comment:
I think in the latest tendermint/sdk, there may be one other fault type defined?